nilearn/datasets/func.py

```diff
@@ -1275,3 +1276,213 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+                                                 standard_variables
+                                                 ):
+        if isinstance(check_in, list):
+            for each_str in check_in:
+                if each_str not in standard:
+                    raise ValueError(message % (
+                        name, check_in, str(standard)))
+                if assign == 'dimensionalities':
+                    dimensionalities = check_in
+                else:
+                    timeseries_methods = check_in
+        elif not isinstance(check_in, list):
+            if check_in not in standard:
+                raise ValueError(message % (
+                    name, check_in, str(standard)))
+
+            if assign == 'dimensionalities':
+                dimensionalities = [check_in]
+            else:
+                timeseries_methods = [check_in]
+
+    output_matrices = ['correlation', 'partial_correlation']
+    if isinstance(matrices, list):
+        for each_type in matrices:
+            if each_type not in output_matrices:
+                raise ValueError(message % ('matrices', matrices,
+                                            output_matrices))
+            output_matrices = matrices
+    elif not isinstance(matrices, list):
+        if matrices not in output_matrices:
+            raise ValueError(message % ('matrices', matrices, output_matrices))
+        output_matrices = [matrices]
+
+    n_combinations = len(dimensionalities) * len(timeseries_methods) * len(output_matrices)
+    dataset_name = 'Megatrawls'
+    files_netmats = []
+    names = []
+    for dim in dimensionalities:
+        filename = os.path.join('3T_Q1-Q6related468_MSMsulc_d' + str(dim))
+        for timeseries in timeseries_methods:
+            dim_timeseries = os.path.join('d' + str(dim) + '_' + str(timeseries))
+            if timeseries == 'multiple_spatial_regression':
+                timeseries = 'ts2'
+            elif timeseries == 'eigen_regression':
+                timeseries = 'ts3'
+            for matrices in output_matrices:
+                dim_timeseries_matrices = os.path.join(dim_timeseries + '_' + matrices)
+                if matrices == 'correlation':
+                    matrices = 'Znet1.txt'
+                elif matrices == 'partial_correlation':
+                    matrices = 'Znet2.txt'
+                each_files_netmats = [(os.path.join(
+                    filename + '_' + str(timeseries), matrices), url, opts)]
+                files_netmats.append(each_files_netmats)
+                names.append(dim_timeseries_matrices)
+
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir, verbose=verbose)
+    description = _get_dataset_descr(dataset_name)
+
+    network_matrices = []
+    for n in range(n_combinations):
+        netmats = _fetch_files(
+            data_dir, files_netmats[n], resume=resume, verbose=verbose)
+        arr = csv_to_array(netmats[0])
+        network_matrices.append(arr)
+
```

Comment:
Several problems in your function:
* you have 2 `for` loops: one to fetch the files, one to load them. They can be merged.
* you have 3 for loops intricated for the 3 parameters. You can factorize that using itertools
* you generate a string for each item of the dataset. That mean that, if I want to have the matrices of dimensionality 20, I'll have to parse your string, which is inconvenient, you can use 3 lists instead.
* there is no reason to call `_fetch_files` several times since you only download one file

So I would write:
```python
    dataset_name = 'Megatrawls'
    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir, verbose=verbose)
    description = _get_dataset_descr(dataset_name)

    # Generate all combinations
    dims, tseries, mats = list(zip(*list(itertools.product(dimensionalities, timeseries, matrices))))
    files = []
    ts_map = dict(multiple_spatial_regression='ts2', eigen_regression='ts3')
    mat_map = dict(correlation='Znet1.txt', partial_correlation='Znet2.txt')
    for dim, tserie, mat in zip(dims, tseries, mats):
        filepath = os.path.join('3T_Q1-Q6related468_MSMsulc_d%d_%s' % (dim, ts_map[tserie]), mat_map[mat])
        files.append((filepath, url, opts))

    # Fetch all files
    files = _fetch_files(data_dir, files, resume=resume, verbose=verbose)

    # Load them
    files = [csv_to_array(f) for f in files]

    # Build the output bunch
    return Bunch(
        description=description,
        dimensionality=dims,
        timeseries=tseries,
        matrices=mats,
        data=files)
```
A good thing can also be to replace the `data` element by `correlation` and `partial_correlation` or both when pertinent.